File: yandex_backend/42. Частый элемент/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class MostFrequentElement
{
    public static int FindMostFrequentElement(List<int> arr)
    {
        Dictionary<int, int> counts = new Dictionary<int, int>();
        int maxCount = 0;
        int mostFrequent = int.MinValue;

        foreach (int num in arr)
        {
            if (!counts.ContainsKey(num))
                counts[num] = 0;

            counts[num]++;

            if (counts[num] > maxCount)
            {
                maxCount = counts[num];
                mostFrequent = num;
            }
            else if (counts[num] == maxCount && num > mostFrequent)
            {
                mostFrequent = num;
            }
        }
        return mostFrequent;
    }

    public static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        List<int> arr = Console.ReadLine().Split().Select(int.Parse).ToList();

        int result = FindMostFrequentElement(arr);
        Console.WriteLine(result);
    }
}

File: yandex_backend/42. Частый элемент/solution_.py
import sys
from collections import defaultdict
from typing import List, Dict

def most_frequent_element(arr: List[int]) -> int:
    counts: Dict[int, int] = defaultdict(int)
    max_count: int = 0
    most_frequent: int = float('-inf')

    for num in arr:
        counts[num] += 1
        if counts[num] > max_count:
            max_count = counts[num]
            most_frequent = num
        elif counts[num] == max_count and num > most_frequent:
            most_frequent = num

    return most_frequent


def main():
    n: int = int(input())
    a: List[int] = list(map(int, input().split()))
    result: int = most_frequent_element(a)
    print(result)


if __name__ == '__main__':
    main()